packages/tooltip/src/vaadin-tooltip-mixin.js

```diff
@@ -467,19 +448,24 @@ export const TooltipMixin = (superClass) =>
 
       this._overlayElement = this.$.overlay;
 
-      this._srLabelController = new SlotController(this, 'sr-label', 'div', {
+      this.__contentController = new SlotController(this, 'overlay', 'div', {
         initializer: (element) => {
           element.id = this._uniqueId;
           element.setAttribute('role', 'tooltip');
-          this._srLabel = element;
+          this.__contentNode = element;
         },
       });
-      this.addController(this._srLabelController);
+      this.addController(this.__contentController);
     }
 
-    /** @private */
-    __computeOpened(manual, opened, autoOpened, connected) {
-      return connected && (manual ? opened : autoOpened);
+    /** @protected */
+    updated(props) {
+      super.updated(props);
+
+      if (props.has('text') || props.has('generator') || props.has('context')) {
+        this.__updateContent();
```

Comment:
I'm not sure about this method. It mainly exists because of an ugly [internal hack](https://vaadin.slack.com/archives/C04JL39LW/p1732275285927089) which is likely used in some customer project and relies on patching `_renderer`. Maybe we can just break that assumption and inline setting `textContent` here in `updated()`.

Reply:
Doesn't really hurt to keep it.